cmake/dependencies/catch.cmake

```diff
@@ -44,6 +55,14 @@ set(openPMD_catch_src ""
     CACHE PATH
     "Local path to Catch2 source directory (preferred if set)")
 
+# tarball fetcher
+set(openPMD_catch_tar "https://github.com/catchorg/Catch2/archive/refs/tags/v2.13.10.tar.gz"
```

Comment:
Will `openPMD_catch_branch`, `openPMD_catch_repo` and `openPMD_catch_src` be removed?

Reply:
We will support both. The order now of precedence is:
- tar, or if `openPMD_catch_tar` is set to [off/false/empty/etc.](https://cmake.org/cmake/help/latest/command/if.html#constant)
- git

Reply:
Docs added that explain the precedence.